### PR TITLE
Lps 154947 Add new functionality to SystemProperties for moving module framework properties next

### PR DESCRIPTION
**Pull request by W2jx**
Hi @tinatian @dantewang ,I created a new ticket that only contains the new functionality of SystemProperties

### CODE REVIEW

**Comment by W2jx** [comment]
Hi @catarineaguiar, could you please help me scan test results? Thanks!

**Comment by catarineaguiar** [comment]
@W2jx code conflicts, rebase please
No regressions found.

**Comment by tinatian** [comment]
@W2jx 
The replace of System.getProperty() is not completed, you need to at least replace the usage for the liferay defined system properties, for the properties like "line.separator", you can keep. 

The reason I accept that both System.getProperty() and SystemProperties.get() is there are some of usages of System.getProperty can not be removed (for example those in petra modules), thus we can not add SF rule to avoid people doing this. 

So let's at least make sure all the system properties defined by liferay is retrieved by SystemProperties.get()

Please resend.

FYI @dantewang

Thanks.
Tina.

File: portal-kernel/src/com/liferay/portal/kernel/util/SystemProperties.java [hunk 1]
```diff
@@ -60,6 +60,10 @@ public static String get(String key) {
 		return value;
 	}
 
+	public static String[] getArray(String key) {
+		return _arrayValues.computeIfAbsent(key, k -> StringUtil.split(get(k)));
+	}
```
Comment (tinatian) [inline portal-kernel/src/com/liferay/portal/kernel/util/SystemProperties.java:81]: @W2jx @dantewang As we are parsing reference at runtime to get the updated value, adding a cache here actually breaks this. I know we have the same cache in PropsUtil so you may not bring any regression here. But the logic itself does not make sense.

Also, we need add test for this.

Tina.

File: portal-kernel/src/com/liferay/portal/kernel/util/SystemProperties.java [hunk 2]
```diff
@@ -220,4 +225,68 @@ public static void set(String key, String value) {
 		load(classLoader);
 	}
 
+	private static class ExtendedProperties extends Properties {
```
Comment (tinatian) [inline portal-kernel/src/com/liferay/portal/kernel/util/SystemProperties.java:277]: @W2jx The logic here is not easy to understand. Let's do this:
1. Read everything from the file and only keep the line which do not lead with "#"
2. Then pass in the extracted result to jdk Properties to load the values

**Comment by tinatian** [comment]
@W2jx Please add unit test to cover the features you added/modified.

FYI @dantewang

Tina.

**Comment by tinatian** [comment]
ci:close

**Pull request closed without merging**